### DISCUSSION TITLE
Running into ArrayIndexOutOfBoundsException issues with PMD

### DISCUSSION THREAD

**Discussion by hareesh1206** [pmd/pmd#4385] [Q&A]
Hello,

I am trying to build a spring-boot project with pmd integrated in gradle and running into ArrayIndexOutOfBoundsException for lot of test cases. Even when I run single test case in debug mode to understand where I am hitting ArrayIndexOutOfBoundsException, by default it is throwing ArrayIndexOutOfBoundsException even before jumping into the class. I am assuming this is due some config or version mismatch or due to missing plugins. can someone please help me what could be the causes for this ? I am using below versions

Jacoco: 0.8.8, pmd: 6.5.4 , jdk 17.   -- Is there any specific compatible versions that I need to use ?

**Comment by adangel** [comment]
Thanks for your question.
Do you see a stacktrace when you run in debug mode? The stacktrace would be helpful to see whether the problem is in PMD, in gradle or somewhere in between.

Also, do you get the exception for a specific source file? If you, could you share this file?

P.S.: I assume, you are using PMD 6.__54__.0 (and not 6.5.4, this doesn't exist)

**Reply by hareesh1206** [reply]
Hi @adangel Thanks for responding to my post. Actually this issue is happening due to jmockit dependency, after using **com.github.hazendaz.jmockit:jmockit** instead of **org.jmockit** the issue is resolved. Thanks.
